src/components/Model3DViewer/Model3DViewer.tsx

```diff
@@ -235,21 +294,157 @@ export class Model3DViewer extends React.Component<Model3DViewerProps> {
     }
   };
 
+  slice = (sliceProps: SlicingProps) => {
+    if (this.viewer) {
+      const planes = this.state.planes;
+      if (sliceProps.x) {
+        const plane = new THREE.Plane(
+          new THREE.Vector3(sliceProps.x.direction ? 1 : -1, 0, 0),
+          sliceProps.x.coord
+        );
+        planes[0] = plane;
+      }
+      if (sliceProps.y) {
+        const plane = new THREE.Plane(
+          new THREE.Vector3(0, sliceProps.y.direction ? 1 : -1, 0),
+          sliceProps.y.coord
+        );
+        planes[1] = plane;
+      }
+      if (sliceProps.z) {
+        const plane = new THREE.Plane(
+          new THREE.Vector3(0, 0, sliceProps.z.direction ? 1 : -1),
+          sliceProps.z.coord
+        );
+        planes[2] = plane;
+      }
+      this.setState({ planes });
+    }
+  };
+
+  takeScreenshot = async () => {
+    if (this.viewer) {
+      const url = await this.viewer.getScreenshot();
+      if (this.props.onScreenshot) {
+        this.props.onScreenshot(url);
+      }
+    }
+  };
+
+  onChange = (val: number, axis: number) => {
```

Comment:
Axis is now a number? It is not obvious that `x` maps to `0`

Reply:
The onChange and flipSlider function are called when sliders are rendered in renderSlider(), so I pass the axis number to renderSlider() when processing the sliders property, and renderSlider can then pass its axis number to onChange and flipSlider.

`const xSlider = this.renderSlider(this.props.slider.x, 0);
  const ySlider = this.renderSlider(this.props.slider.y, 1);
  const zSlider = this.renderSlider(this.props.slider.z, 2);`
`renderSlider = (range: SliderRange | undefined, axisNumber: number) => {
...
this.onChange(val, axisNumber)
}`